src/libraries/System.Text.Json/Common/ReflectionExtensions.cs

```diff
@@ -229,5 +230,85 @@ public static bool IsVirtual(this PropertyInfo? propertyInfo)
             Debug.Assert(propertyInfo != null);
             return propertyInfo != null && (propertyInfo.GetMethod?.IsVirtual == true || propertyInfo.SetMethod?.IsVirtual == true);
         }
+
+        public static bool IsKeyValuePair(this Type type, Type? keyValuePairType = null)
```

Comment:
Nit: not a huge fan of exposing internal helper methods as extension methods.

Reply:
This just moves the helper to a shared file with the source generator which also needs the functionality. Other shared helpers are written extension methods..